core/src/main/java/nl/nn/adapterframework/doc/ModelBuilder.java

```diff
@@ -0,0 +1,265 @@
+		}
+		if(!interfaze.isInterface()) {
+			throw new IllegalArgumentException("This exists on the classpath but is not an interface: " + interfaceName);
+		}
+		Set<SpringBean> unfiltered = InfoBuilderSource.getSpringBeans(interfaze);
+		List<SpringBean> result = new ArrayList<SpringBean>();
+		for(SpringBean b: unfiltered) {
+			if(interfaze.isAssignableFrom(b.getClazz())) {
+				result.add(b);
+			}
+		}
+		return result;
+	}
+
+	FrankDocGroup addGroup(String name) {
+		FrankDocGroup group = new FrankDocGroup(name);
+		group.setElements(new HashMap<>());
+		model.getGroups().add(group);
+		return group;
+	}
+
+	void addElementsToGroup(Class<?> clazz, FrankDocGroup group) {
+		group.getElements().put(clazz.getName(), frankElement(clazz));
+	}
+
+	FrankElement frankElement(Class<?> clazz) {
+		if(model.getAllElements().containsKey(clazz.getName())) {
+			return model.getAllElements().get(clazz.getName());
+		}
+		List<Class<?>> classesForNewElements = new ArrayList<>();
+		FrankElement parent = null;
+		Class<?> superClass = clazz;
+		while(true) {
+			classesForNewElements.add(superClass);
+			superClass = superClass.getSuperclass();
+			if(superClass == null) {
+				break;
+			}
+			if(model.getAllElements().containsKey(superClass.getName())) {
+				parent = model.getAllElements().get(superClass.getName());
+				break;
+			}
+		}
+		Collections.reverse(classesForNewElements);
+		for(Class<?> seed: classesForNewElements) {
+			parent = createFrankElement(seed, parent);
+			model.getAllElements().put(parent.getFullName(), parent);
+		}
+		return parent;		
+	}
+
+	FrankElement createFrankElement(Class<?> clazz, FrankElement parent) {
+		FrankElement result = new FrankElement(clazz.getName(), clazz.getSimpleName());
+		result.setParent(parent);
+		result.setAttributes(createAttributes(result, clazz.getDeclaredMethods()));
+		return result;
+	}
+
+	List<FrankAttribute> createAttributes(FrankElement frankElement, Method[] methods) {
+		Map<String, String> setterAttributes = getAttributeToMethodNameMap(methods, "set");
+		Map<String, String> getterAttributes = getAttributeToMethodNameMap(methods, "get");
+		getterAttributes.putAll(getAttributeToMethodNameMap(methods, "is"));
+		Map<String, String> setterToAttributeName = new HashMap<>();
+		for(String candidateAttributeName: setterAttributes.keySet()) {
+			setterToAttributeName.put(setterAttributes.get(candidateAttributeName), candidateAttributeName);
+		}
+		List<FrankAttribute> result = new ArrayList<>();
+		for(Method method: methods) {
+			if(setterToAttributeName.containsKey(method.getName())) {
+				String candidateAttributeName = setterToAttributeName.get(method.getName());
+				FrankAttribute candidate = new FrankAttribute(candidateAttributeName);
+				candidate.setDescribingElement(frankElement);
+				boolean isDocumented = documentAttribute(candidate, method);
+				if(getterAttributes.containsKey(candidateAttributeName) || isDocumented) {
```

Comment:
Waarom test je of er ook een getter is bij de setter? Dat is toch niet nodig?
Je zou hoogstens een warning kunnen geven als er een getter is van een ander type dan de setter, want dat geeft problemen bij het laden van de configuratie.
En ook nog: als je zo'n test doet zoals hier, dan moet je niet eerst je hele candidate opbouwen en daarna pas checken of je hem wilt hebben, maar eerst die check doen.

Reply:
De check of er een getter is neem ik over van de bestaande code. Ik heb aangenomen dat als er alleen een setter is zonder annotaties, dat die dan niet bedoeld is als attribute. Als ik deze check weglaat, dan krijgen we misschien meer attributes dan er in de huidige Frank!Doc zijn.

Reply:
Over het bestaan van candidates die geen attributes hoeven te worden. Ik ging ervan uit dat een setter alleen geen attribute wordt. Maar je hebt zeker een attribute als er een @IbisDoc of @IbisDocRef op staat. Die annotaties moeten geparsed voordat je beslist of er sprake is van een attribute. Maar de data in de @IbisDoc annotatie hoort in een FrankAttribute. Het is dus handig om al een FrankAttribute te vullen bij het parsen van @IbisDoc. Wat vind je daarvan?

Reply:
> De check of er een getter is neem ik over van de bestaande code. Ik heb aangenomen dat als er alleen een setter is zonder annotaties, dat die dan niet bedoeld is als attribute. Als ik deze check weglaat, dan krijgen we misschien meer attributes dan er in de huidige Frank!Doc zijn.

Het is het beste om het volgende te doen:
1) check of de getter van het zelfde type is als het argument van de setter. Als dat niet zo is, geef dan een waarschuwing aan de Core developer
2) als er een setter is met een primitive type (string, int, boolean, ...) argument, zonder @IbisDoc of @IbisDocRef, geef dan een waarschuwing aan de Core developer. Voor het geval dat we echt geen documentatie willen, kan er een @NoIbisDoc annotatie komen. Als en zolang die @NoIbisDoc er niet is komt het attribuut wel in de ibisdoc, zonder beschrijving.

>  Het is dus handig om al een FrankAttribute te vullen bij het parsen van @IbisDoc. 

Dat lijkt me inderdaad de goede manier

Reply:
Gedaan.